filesystems/gcs/src/main/scala/cromwell/filesystems/gcs/GcsPathBuilder.scala

```diff
@@ -85,10 +85,22 @@ object GcsPathBuilder {
       if (uri.getScheme == null) PossiblyValidRelativeGcsPath
       else if (uri.getScheme.equalsIgnoreCase(CloudStorageFileSystem.URI_SCHEME)) {
         if (uri.getHost == null) {
-          softBucketParsing(string) map { ValidFullGcsPath(_, uri.getPath) } getOrElse InvalidFullGcsPath(string)
-        } else ValidFullGcsPath(uri.getHost, uri.getPath)
-      } else InvalidScheme(string)
-    } recover { case t => UnparseableGcsPath(string, t) } get
+          softBucketParsing(string) map {
+            ValidFullGcsPath(_, uri.getPath)
+          } getOrElse {
+            InvalidFullGcsPath(string)
+          }
+        } else if (Option(uri.getPath).forall(_.trim.nonEmpty)) {
```

Comment:
what do these new cases represent?

Reply:
- The path is null
- The path is empty
- The path is just whitespace

As Chris pointed it out I need to test/fix my boolean logic swapping `forall` with `exists`